node/forest_libp2p/src/behaviour.rs

```diff
@@ -164,10 +164,15 @@ impl NetworkBehaviourEventProcess<BitswapEvent> for ForestBehaviour {
 
 impl NetworkBehaviourEventProcess<GossipsubEvent> for ForestBehaviour {
     fn inject_event(&mut self, message: GossipsubEvent) {
-        if let GossipsubEvent::Message(_, _, message) = message {
+        if let GossipsubEvent::Message {
+            propagation_source,
+            message,
+            message_id: _,
+        } = message
+        {
             self.events.push(ForestBehaviourEvent::GossipMessage {
-                source: message.source,
-                topics: message.topics,
+                source: propagation_source,
+                topic: message.topic,
                 message: message.data,
```

Comment:
I think the `message` struct member should be destructured, and then restructured here. It'd make the code less verbose, sorta like this (starting on line 169):

```rust
message: ForestBehaviourEvent::GossipMessage {
    source,
    topics,
    data,
}
```